### DISCUSSION TITLE
Faster Download of Single Large Files through Multithreaded Approach

### DISCUSSION THREAD

**Discussion by valentinneher** [icloud-photos-downloader/icloud_photos_downloader#936] [Ideas]
I'd like to be able to download large files (eg. Video Files >10GB) faster than using icloud.com.
If this is already implemented, I apologise.

**Comment by AndreyNikiforov** [comment]
`icloudpd` cannot download files faster than icloud.com. The following limits apply:
- your internet connection
- Apple rate limits if any

When you are downloading from icloud.com with a browser you are hitting the same limits. Downloading with multiple streams (e.g. multithreaded) will not overcome these limit.

`icloudpd` does not have a capability to filter download by their size in bytes.

Consider another approach - have `icloudpd` constantly running and synchronizing your iCloud collection with local storage. Your focus will be on the lag between assets appearing in iCloud and on your storage; and speed of download would not be a major factor any more.
